ckan/config/config_declaration.yaml

```diff
@@ -408,20 +408,88 @@ groups:
           token. This is useful in some scenarios where using the default ``Authorization`` one
           causes problems.
 
-      - key: ckan.cache_expires
-        default: 0
+      - key: ckan.cache.expires
+        default: 3600
+        legacy_key: ckan.cache_expires  # since v.2.12.0
         type: int
         example: 2592000
-        description: This sets ``Cache-Control`` header's max-age value.
+        description: |
+          This sets ``Cache-Control`` header's max-age value on public cache.
+          3600 = 1 hour before revalidate with shared cache
+
+      - key: ckan.cache.shared.expires
+        default: 7200
+        type: int
+        example: 7200
+        description: |
+          This sets ``Cache-Control`` header's s-maxage value on public cache.
+          7200 = 2 hour before shared cache revalidates
+
+      - key: ckan.cache.private.expires
+        type: int
+        default: 60
+        example: 3600
+        description: |
+          This sets ``Cache-Control`` header's max-age value on private cache.
 
-      - key: ckan.cache_enabled
+      - key: ckan.cache.public.enabled
+        legacy_key: ckan.cache_enabled  # since v.2.12.0
+        default: True
         type: bool
         example: "true"
         description: |
           This enables cache control headers on all requests. If the user is
           not logged in and there is no session data a ``Cache-Control:
-          public`` header will be added. For all other requests the
-          ``Cache-control: private`` header will be added.
+          public`` header will be added. For all other requests see `
+          `ckan.cache.private.enabled`` config option.
+
+      - key: ckan.cache.private.enabled
+        default: true
+        type: bool
+        example: "true"
+        description: |
+          When this is set to true and cache type requested is private,
+          then ``Cache-control: private`` header will be added,
+          else ``Cache-Control: no-cache max-age=0`` will be set on all
+          requests except for sensitive responses where no-store
+          will also be set.
+
+      - key: ckan.cache.stale_while_revalidates
+        default: 3600
```

Comment:
@wardi  we may also want to put in migration notes to enable existing behaviour to set both ``ckan.cache.stale_while_revalidates`` and ``ckan.cache.stale_if_error`` to 0 so ``must-revalidate`` is sent like it does currently. 

Enabling this options allows pages to be background refresh and when loss of internet allows pages to not be 'nuked' if cache has expired and can't be retrieved.